### ISSUE TITLE
Limit price should be optional

### ISSUE THREAD

**Issue by enriquejr99** [BitMEX/api-connectors#377]
The price variable should not be required as specified in the [docs](https://www.bitmex.com/api/explorer/#!/Order/Order_new ) when creating limit orders. Although this may seem strange, it allows traders to create buy limit orders at the bid price and sell limit orders at the ask price which can help reduce the total commission paid over the long run.

However, not setting the price currently throws an error when creating a limit order. I have not tested this with other order operations, such as bulk, but it is likely it will also fail.

Due to high volatility, the current bid price may change while a trader sends a new limit order to the server. If this happens, the trader may end up paying a high fee as the limit order will be executed at the market price. Example:

1. Server sends Alice the current orderbook.
2. Alice gets the highest bid from the orderbook and sends a new limit buy order to the Server using the highest bid price as the limit price.
3. While the new order reaches the Server, Bob sells many contracts and pushes the price down.
4. Alice's new order reaches the Server with a limit price that can now be executed at market price and hence Alice is charged a 0.075% fee.

The solution; which Binance uses, is to NOT execute the limit order on market and instead return an error. This prevents this kind of unfortunate lag situations.

In conclusion, the limit price argument should be optional as intended in the docs and limit orders should not be executed on market, as they weren't intended to do so. Fixing any of the two will solve this issue. Fixing both would be awesome.

**Comment by enriquejr99** [comment]
Just realized we can use `execInst='ParticipateDoNotInitiate'` to prevent the limit order from being market sold. I suggest using this execInst value as default for all limit orders.

**Comment by enriquejr99** [comment]
Another thing to note here is that although using `execInst='ParticipateDoNotInitiate'` prevents an order from being market sold on creation, it does not prevent it from being market sold on amendment. This condition should persist on amends too.
